frontend/src/metabase/components/form/CustomForm/CustomFormSubmit.tsx

```diff
@@ -65,6 +74,49 @@ CustomFormSubmitLegacyContext.contextTypes = _.pick(
   "submitTitle",
   "renderSubmit",
   "disablePristineSubmit",
+  "shouldPersistError",
+  "fields",
+  "error",
 );
 
+interface IsFormValidProps {
+  fields: NestedFormField;
+  invalid: boolean;
+  shouldPersistError?: boolean;
+  error?: string;
+}
+
+function isFormValid({
+  shouldPersistError,
+  fields,
+  invalid,
+  error,
+}: IsFormValidProps): boolean {
+  if (shouldPersistError && error) {
+    return areAllFieldsUntouched(fields);
```

Comment:
Only use `touched` state to determine whether submit button should be disabled when there is an error. Because `invalid` will be true while we're fixing the form.